tests/dags/providers/provider_api_scripts/test_inaturalist.py

```diff
@@ -0,0 +1,101 @@
+from ast import literal_eval
+from pathlib import Path
+
+import pytest
+from common.licenses import get_license_info
+from providers.provider_api_scripts import inaturalist
+
+
+# This file supports testing of the customized functions within the provider ingestion
+# class. It does not cover the pre-ingestion steps or sqls. To test those, instantiate
+# the local dev environment, with the inaturalist-open-data bucket on the minio docker.
+# (`just recreate` should do the trick.) Then manually kick off the Airflow DAG, and
+# inspect the logs and the data saved to openverse s3 on minio.
+
+# Based on the small sample files in /tests/s3-data/inaturalist-open-data, expect:
+# taxa.csv.gz --> 183 records
+# observations.csv.gz ---> 31 records
+# observers.csv.gz ---> 22 records
+# photos.csv.gz ---> 36 records
+# 3 of the photo records link to unclassified observations, so they have no title or
+# tags and we don't load them.
+# 10 of the remaining photo ids appear on multiple records, load one per photo_id.
+# The expected response to the database query for transformed data is a list of tuples
+# (not technically json, though the first value in each tuple is a dict) in
+# ./resources/inaturalist/full_db_response.txt
+
+
+INAT = inaturalist.iNaturalistDataIngester()
+RESOURCE_DIR = Path(__file__).parent / "resources/inaturalist"
+FULL_DB_RESPONSE = literal_eval((RESOURCE_DIR / "full_db_response.txt").read_text())
```

Comment:
TIL about `literal_eval` as well :thinking: I would have probably reached for `json.loads` or maybe even pickle if I wanted to keep it in native Python for some reason. Interesting!

Reply:
I did try `json.loads`, but the database always returns a list of tuples (even if there's only one tuple with only one value), which doesn't exist in json format. So `literal_eval` seemed easiest. But do you think I should put that in a comment and/or try harder with `json.loads` or `pickle`?

Reply:
Ahh, no definitely no need! Maybe an explanatory comment to document why it's done that way than using a more typical format, but this is a good readable way to store and solve that problem, from my perspective. I was truly just noting a curiosity for me here, not a suggestion to change anything :)

Reply:
I was thinking more about this last night and it gave me an idea / push to try again with json on the postgres side, and I think I may have come up with a much better approach as a result. So thank you! I'm going to just do a couple of more tests tonight and then probably commit an update.

Reply:
I've used `literal_eval` a bunch in the past for exactly the case you're describing Rebecca! It's useful for re-hydrating things like `set`s as well, since there isn't a JSON type that represents them either.

Reply:
OK, so while I was thinking about all of this, I realized that I really could do better with the json conversion within postgres, and while it doesn't get rid of `literal_eval` for testing purposes (what does hydrating a set mean?), it does really simplify the data ingester. (And fix a bug with no-tag photos.) So that's the last two commits I added today.